frmts/hdf5/bagdataset.cpp

```diff
@@ -2215,11 +2215,11 @@ bool BAGDataset::OpenRaster(GDALOpenInfo* poOpenInfo,
     else if( !osGeorefMetadataLayer.empty() )
     {
         auto poGeoref_metadataLayer = m_poRootGroup->OpenGroupFromFullname(
-            "/BAG_root/Georef_metadata/" + osGeorefMetadataLayer, nullptr);
+            "/BAG_root/georef_metadata/" + osGeorefMetadataLayer, nullptr);
```

Comment:
I am not aware of any way to write this layer other than hacking it directly in HDF5 until now (with the BAG library update).  GDAL only provided read access (correct?).  Since this layer only became a finalized part of the BAG specification this week, no BAGs would exists that would comply with any formalized specification.  I can't imagine someone writing this layer with the old path and expecting it to be supported.  I think simplicity in the library is better rather than adding complexity to support an unlikely and unspecified file.

Reply:
> I think simplicity in the library is better rather than adding complexity to support an unlikely and unspecified file.

ok thanks for the context. Sometimes it happens that draft specs languish for years before being finalized and that significant amount of data is yet released in the wild against those draft versions, and it can thus make sense to support the old non-official way. But I understand we're not in that situation.